### PR TITLE
Handle nested nils

### PR DESCRIPTION
**Pull request by sco11morgan**

### CODE REVIEW

File: lib/deep_slice.rb [hunk 1]
```diff
@@ -13,9 +13,19 @@ def deep_slice(*keys)
             recursive_keys = key[sub_key]
             result[sub_key] = \
               if self[sub_key].is_a?(Array)
-                self[sub_key].map { |val| val.deep_slice(*[recursive_keys].flatten) }
+                self[sub_key].map { |val|
```
Comment (jiacona) [inline lib/deep_slice.rb:16]: Might be a bit tidier if written like `val.deep_slice(*[recursive_keys].flatten) unless val.nil?`

File: lib/deep_slice.rb [hunk 2]
```diff
@@ -13,9 +13,19 @@ def deep_slice(*keys)
...
+                  if val.nil?
+                    nil
+                  else
+                    val.deep_slice(*[recursive_keys].flatten)
+                  end
+                }
               else
-                 self[sub_key].deep_slice(*[recursive_keys].flatten)
+                 if self[sub_key].nil?
```
Comment (jiacona) [inline lib/deep_slice.rb]: Same here, I would prefer `self[sub_key].deep_slice(*[recursive_keys].flatten) unless self[sub_key].nil?`

**Review by jiacona — CHANGES_REQUESTED** [review]
Thanks for the PR. I just have a few comments but will be happy to merge if addressed

**Comment by schmidtg** [comment]
:shipit:

**Comment by sco11morgan** [comment]
@jiacona done

**Review by jiacona — APPROVED** [review]
Looks good! Thanks again for the contribution

**Comment by jiacona** [comment]
Hah, I just realized I'm not actually a committer on this repo anymore. @schmidtg care to help out?

**Comment by schmidtg** [comment]
Sure! Thanks @jiacona and @sco11morgan for adding the `nil` option with supported tests.

@sco11morgan , do you mind cleaning up the commits (merge into one commit and write appropriate message for what you're introducing) to keep the git history a little tidier?

Thanks

**Comment by sco11morgan** [comment]
@schmidtg done.  can you publish the gem after merging?

**Comment by jiacona** [comment]
@sco11morgan, 0.1.3 is up on rubygems

**Pull request merged**